rch/src/main/java/org/motechproject/nms/rch/service/impl/RchWebServiceFacadeImpl.java

```diff
@@ -662,71 +662,82 @@ public void readVillageResponseFromFile(MotechEvent event) throws RchFileManipul
+
+                        if (mothersDataSet == null || mothersDataSet.getRecords() == null) {
+                            String warning = String.format("No mother data set received from RCH for %s state", stateName);
                             LOGGER.warn(warning);
+                            rchImportAuditDataService.create(new RchImportAudit(startDate, endDate, RchUserType.MOTHER, stateCode, stateName, 0, 0, warning));
+                        } else {
+                            LOGGER.info("Received {} mother records from RCH for {} state", sizeNullSafe(mothersDataSet.getRecords()), stateName);
+
+                            RchImportAudit audit = saveImportedMothersData(mothersDataSet, stateName, stateCode, startDate, endDate);
+                            rchImportAuditDataService.create(audit);
+                            stopWatch.stop();
+                            double seconds = stopWatch.getTime() / THOUSAND;
+                            LOGGER.info("Finished RCH mother import dispatch in {} seconds. Accepted {} mothers, Rejected {} mothers",
+                                    seconds, audit.getAccepted(), audit.getRejected());
+
+                            LOGGER.info("fromDate for delete {} {}", startDate, endDate);
+                            deleteRchImportFailRecords(startDate, endDate, RchUserType.MOTHER, stateId);
                         }
-                    } catch (JAXBException e) {
-                        throw new RchInvalidResponseStructureException(String.format("Cannot deserialize RCH mother data from %s location.", stateId), e);
-                    } catch (RchInvalidResponseStructureException e) {
-                        String error = String.format("Cannot read RCH mothers data from %s state with stateId: %d. Response Deserialization Error", stateName, stateId);
-                        LOGGER.error(error, e);
-                        alertService.create(RCH_WEB_SERVICE, "RCH Web Service Mother Import", e
-                                .getMessage() + " " + error, AlertType.CRITICAL, AlertStatus.NEW, 0, null);
-                        rchImportAuditDataService.create(new RchImportAudit(startDate, endDate, RchUserType.MOTHER, stateCode, stateName, 0, 0, error));
-                        rchImportFailRecordDataService.create(new RchImportFailRecord(endDate, RchUserType.MOTHER, stateId));
-                    } catch (NullPointerException e) {
-                        LOGGER.error("No files saved a : ", e);
+                    } else {
+                        String warning = String.format("No Mother data set received from RCH for %d stateId", stateId);
+                        LOGGER.warn(warning);
                     }
+                } catch (JAXBException e) {
+                    throw new RchInvalidResponseStructureException(String.format("Cannot deserialize RCH mother data from %s location.", stateId), e);
+                } catch (RchInvalidResponseStructureException e) {
```

Comment:
there should never be such space difference in the commits !! tough to review ,. Please do git diff before ccommit